httpreplay/cobweb.py

```diff
@@ -173,5 +187,214 @@ def handle(self, s, ts, protocol, sent, recv):
             # or TLS stream straight ahead to our parent.
             self.parent.handle(s, ts, protocol, sent, recv)
 
+
 class SmtpProtocol(Protocol):
     """Interprets the SMTP protocol."""
+
+    _commands = [
+        "ehlo", "helo", "mail from", "rcpt to", "etrn",
+        "turn", "atrn", "size", "etrn", "pipelining",
+        "chunking", "data", "dsn", "rset", "vrfy",
+        "help", "quit", "noop", "expn", "auth login",
+        "auth plain", "auth cram-md5", "auth",
+        "binarymime", "relay", "size", "starttls", "checkpoint",
+        "enhancedstatuscodes", "8bitmime", "send"
+    ]
+
+    _command_field = {
+        "ehlo": "hostname",
+        "helo": "hostname",
+        "mail from": "mail_from",
+        "rcpt to": "mail_to"
+    }
+
+    # Min to max SMTP response codes
+    _min = 100
+    _max = 600
+
+    def handle(self, s, ts, protocol, sent, recv):
+        if protocol != "tcp":
+            self.parent.handle(s, ts, protocol, sent, recv)
+            return
+
+        if self.stream is None:
+            self.stream = self.parent.tcp.streams[s]
+
+        self.parse_request(sent)
+        self.parse_reply(recv)
+
+        if self.stream.state in ["conn_finish", "conn_closed"]:
+            self.parent.handle(s, ts, "smtp", self.request, self.reply)
+
+    def init(self, *args, **kwargs):
+        self.request = SmtpRequest()
+        self.reply = SmtpReply()
+
+        # Current smtp command from client
+        self.command = None
+
+        # Last smtp server response code
+        self.last_rescode = 0
+
+        # Last smtp server response message
+        self.last_resmess = ""
+        self.stream = None
+
+    def _split(self, data, spliton="\r\n", always_list=False, maxsplit=-1):
+        splitdata = filter(None, data.split(spliton, maxsplit))
+        if len(splitdata) < 1:
+            return None
+        elif len(splitdata) == 1 and not always_list:
+            return splitdata[0]
+        else:
+            return splitdata
```

Comment:
Based on its usage, I'd say that this function is unnecessary. Please replace it with simple `s.split()` statements etc :)
Note that you can do `s.split(None)` to split on anything whitespace'ish.